stock_rule_reserve_max_quantity/tests/test_stock_rule_reservation.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+
+from odoo.addons.stock.tests.common import TestStockCommon
+
+
+class TestStockRuleReservation(TestStockCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_tv = cls.env["product.product"].create(
+            {
+                "name": "Product Variable QTYs",
+                "type": "product",
+                "categ_id": cls.env.ref("product.product_category_all").id,
+            }
+        )
+
+    def _enable_pick_ship(self, create_backorder="never"):
+        """Enable pick_ship route and set the pick rule to reserve_max_quantity"""
+        self.wh = self.env["stock.warehouse"].search(
+            [("company_id", "=", self.env.user.id)], limit=1
+        )
```

Comment:
issue: avoid polluting other tests:

```suggestion
        self.wh = self.env["stock.warehouse"].search(
            [("company_id", "=", self.env.user.id)], limit=1
        )
        self.addCleanup(delattr, self, "wh")
```

Please do the same with other attributes set in this method.

OTOH, you could just do all this stuff in `setUpClass` which would be more performant. After all, all tests need it.